packages/ember-data/lib/system/store.js

```diff
@@ -1839,12 +1845,11 @@ Store = Service.extend({
   adapterFor: function(modelOrClass) {
     var modelName;
 
-    Ember.deprecate(`Passing classes to store methods has been removed. Please pass a dasherized string instead of ${Ember.inspect(modelName)}`, typeof modelOrClass === 'string');
-
-    if (typeof modelOrClass !== 'string') {
-      modelName = modelOrClass.modelName;
-    } else {
+    if (typeof modelOrClass === 'string') {
       modelName = modelOrClass;
+    } else {
+      Ember.deprecate(`Passing classes to store methods has been removed. Please pass a dasherized string instead of ${Ember.inspect(modelName)}`, false, { id: 'ds.store.passing-classes-deprecated', until: '3.0.0' });
```

Comment:
do these ids need to be totally unique? Like, can this id be the same as the one in `serializerFor`?